system/security/cryptohelper/pke.cpp

```diff
@@ -102,39 +115,69 @@ class CLoadedPublicKeyFromMemory : public CLoadedKey
 
 class CLoadedPrivateKeyFromFile : public CLoadedKey
 {
-public:
-    CLoadedPrivateKeyFromFile(const char *keyFile, const char *passPhrase)
+private:
+    MemoryBuffer passPhraseMB;
+    void load(const char *keyFile, const char *passPhrase)
     {
         if (!loadKeyFromFile(keyFile))
             throwEVPException(0, "loadKeyFromFile: failed to open private key");
-        RSA *rsaKey = PEM_read_bio_RSAPrivateKey(keyBio, nullptr, nullptr, (void*)passPhrase);
+        RSA *rsaKey;
+        if (passPhraseMB.length())
+            rsaKey= PEM_read_bio_RSAPrivateKey(keyBio, nullptr, passphraseCB, (void *)&passPhraseMB);//Binary passphrase
+        else
+            rsaKey= PEM_read_bio_RSAPrivateKey(keyBio, nullptr, nullptr, (void*)passPhrase);//plain text or no passphrase
         if (!rsaKey)
             throwEVPException(0, "Failed to create private key");
         finalize(rsaKey, keyFile);
     }
+public:
+    CLoadedPrivateKeyFromFile(const char *keyFile, size32_t passPhraseLen, const void *passPhrase)
+    {
+        passPhraseMB.append(passPhraseLen, passPhrase);
+        load(keyFile, nullptr);
+    }
+    CLoadedPrivateKeyFromFile(const char *keyFile, const char *passPhrase)
+    {
+        load(keyFile, passPhrase);
+    }
 };
 
 class CLoadedPrivateKeyFromMemory : public CLoadedKey
 {
-public:
-    CLoadedPrivateKeyFromMemory(const char *key, const char *passPhrase)
+private:
+    MemoryBuffer passPhraseMB;
+    void load(const char *key, const char *passPhrase)
     {
         loadKeyFromMem(key);
-        RSA *rsaKey = PEM_read_bio_RSAPrivateKey(keyBio, nullptr, nullptr, (void*)passPhrase);
+        RSA *rsaKey;
+        if (passPhraseMB.length())
+            rsaKey= PEM_read_bio_RSAPrivateKey(keyBio, nullptr, passphraseCB, (void *)&passPhraseMB);//Binary passphrase
+        else
+            rsaKey= PEM_read_bio_RSAPrivateKey(keyBio, nullptr, nullptr, (void*)passPhrase);//plain text or no passphrase
         if (!rsaKey)
             throwEVPException(0, "Failed to create private key");
         finalize(rsaKey, "<inline>");
     }
+public:
+    CLoadedPrivateKeyFromMemory(const char *key, size32_t passPhraseLen, const void *passPhrase)
+    {
+        passPhraseMB.append(passPhraseLen, passPhrase);
```

Comment:
I suspect it would be better to combine CLoadedPrivateKeyFromFile and CLoadedPrivateKeyFromMemory, as they duplicate code. The ctor share same prototypes but could differentiate with a bool.
Or could use a CLoadedPrivateKeyBase class.